packages/website/utils/content.ts

```diff
@@ -86,18 +44,57 @@ async function fetchFiles(targetPath: string) {
   return fetchedFiles;
 }
 
-async function geltAllMDX() {
-  const allMDX = await fetchFiles(componentsPath);
-  const mdxData = allMDX.map((filepath) => getSlugByFilepath(filepath));
+async function getAllMdx(paths: string[]) {
+  let allMDX: string[] = [];
 
-  // Saving a map of slugs and the filepath to their mdx in a JSON file
-  const mdxFilepathBySlug = mdxData.reduce((acc, mdx) => {
-    return { ...acc, [mdx.slug]: mdx.filepath };
-  }, {});
-  const data = JSON.stringify(mdxFilepathBySlug, null, 2);
-  fs.writeFileSync('utils/mdxFilepathBySlug.json', data);
+  for (const path of paths) {
+    const newFiles = await fetchFiles(path);
+    allMDX = allMDX.concat(newFiles);
+  }
 
+  const mdxData = allMDX
+    .map((filepath) => getDataByFilepath(filepath))
+    .filter((data) => {
+      // return only those mdx files which have 'slug' set in meta section
+      return data.frontMatter.data.slug;
+    });
   return mdxData;
 }
 
-export { geltAllMDX, getPageBySlug };
+const allMdxSources = [
+  path.resolve('../../packages/components'),
+  path.resolve('../../packages/core'),
+  path.resolve('../../packages/forma-36-website/src/content'),
+];
+
+async function getMdxSourceBySlug(slug: string[]) {
+  const mdxFiles = await getAllMdx(allMdxSources);
```

Comment:
Doesn't this mean that now every time I go to a page I will get allMdx?

In the previous approach with the json file, we didn't need to do that
when you go to a page you just read the json to discover where the file of your slug is

it seems unnecessary to call getAllMdx for each page

Reply:
It's called once during the build and doesn't affect performance. No runtime implications either.